src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java

```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
+	@Test
+	public void testGetSecurityGroupForPublicIpNetworkNull() throws UnauthorizedRequestException, InstanceNotFoundException,
+			InvalidParameterException {
+		// set up
+		Mockito.when(this.virtualNetwork.xpath(Mockito.anyString())).thenReturn(null);
 
 		// exercise
-		this.plugin.doDeleteInstance(client, this.publicIpOrder);
+		SecurityGroup secGroup = this.plugin.getSecurityGroupForPublicIpNetwork(this.client, this.instanceId);
 
 		// verify
-		Mockito.verify(response, Mockito.times(1)).getErrorMessage();
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.getVirtualNetwork(Mockito.eq(this.client), Mockito.eq(this.instanceId));
+
+		Mockito.verify(this.virtualNetwork, Mockito.times(TestUtils.RUN_ONCE)).xpath(Mockito.eq(VNET_TEMPLATE_SECURITY_GROUPS_PATH));
+		Assert.assertNull(secGroup);
 	}
-	
-	// test case: When calling the deleteSecurityGroup method and not get a
-	// satisfactory response from the cloud, the resource will not be removed.
+
+	// test case: when invoking doDeleteInstance the plugin should retrieve the respective ONe
+	// virtual network and delete it
 	@Test
-	public void testDeleteSecurityGroupUnsuccessfully() throws UnexpectedException, UnauthorizedRequestException,
-			InvalidParameterException, InstanceNotFoundException {
+	public void testDoDeleteInstance() throws UnexpectedException, InstanceNotFoundException, InvalidParameterException,
+			UnauthorizedRequestException {
 		// set up
-		Client client = Mockito.mock(Client.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		BDDMockito.given(OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString()))
-				.willReturn(client);
+		Mockito.when(this.virtualNetwork.delete()).thenReturn(this.response);
+		Mockito.when(this.response.isError()).thenReturn(false);
 
-		String securityGroupId = STRING_ID_ONE;
-		SecurityGroup securityGroup = Mockito.mock(SecurityGroup.class);
-		BDDMockito.given(OpenNebulaClientUtil.getSecurityGroup(Mockito.eq(client), Mockito.eq(securityGroupId)))
-				.willReturn(securityGroup);
+		// exercise
+		this.plugin.doDeleteInstance(client, this.publicIpOrder);
 
-		OneResponse response = Mockito.mock(OneResponse.class);
-		Mockito.when(securityGroup.delete()).thenReturn(response);
-		Mockito.when(response.isError()).thenReturn(true);
+        // verify
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.getVirtualNetwork(Mockito.eq(this.client), Mockito.eq(this.instanceId));
+
+		Mockito.verify(this.virtualNetwork, Mockito.times(TestUtils.RUN_ONCE)).delete();
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).isError();
+	}
+
+	// test case: when invoking doDeleteInstance with invalid parameters the plugin should
+	// throw an UnexpectedException
+	@Test
+	public void testDoDeleteInstanceFail() throws InstanceNotFoundException, InvalidParameterException,
```

Comment:
This test has a runtime error...
Please check and correct.

Reply:
done